boards/imxrt1050-evkb/src/main.rs

```diff
@@ -0,0 +1,438 @@
+//! Reference Manual for the Imxrt-1052 development board
+//!
+//! - <https://www.nxp.com/webapp/Download?colCode=IMXRT1050RM>
+
+#![no_std]
+#![no_main]
+#![deny(missing_docs)]
+
+use capsules::virtual_alarm::{MuxAlarm, VirtualMuxAlarm};
+use components::gpio::GpioComponent;
+use kernel::capabilities;
+use kernel::common::dynamic_deferred_call::{DynamicDeferredCall, DynamicDeferredCallClientState};
+use kernel::component::Component;
+use kernel::debug;
+use kernel::hil::gpio::Configure;
+use kernel::hil::time::Alarm;
+use kernel::Platform;
+use kernel::{create_capability, static_init};
+
+// use components::fxos8700::Fxos8700Component;
+// use components::ninedof::NineDofComponent;
+
+use imxrt1050::iomuxc::DriveStrength;
+use imxrt1050::iomuxc::MuxMode;
+use imxrt1050::iomuxc::OpenDrainEn;
+use imxrt1050::iomuxc::PadId;
+use imxrt1050::iomuxc::PullKeepEn;
+use imxrt1050::iomuxc::PullUpDown;
+use imxrt1050::iomuxc::Sion;
+use imxrt1050::iomuxc::Speed;
+
+// Unit Tests for drivers.
+// #[allow(dead_code)]
+// mod virtual_uart_rx_test;
+
+/// Support routines for debugging I/O.
+pub mod io;
+
+/// Defines a vector which contains the boot section
+pub mod boot_header;
+
+// Number of concurrent processes this platform supports.
+const NUM_PROCS: usize = 1;
+
+// Actual memory for holding the active process structures.
+static mut PROCESSES: [Option<&'static dyn kernel::procs::ProcessType>; NUM_PROCS] = [None];
+
+static mut CHIP: Option<&'static imxrt1050::chip::Imxrt1050> = None;
+
+// How should the kernel respond when a process faults.
+const FAULT_RESPONSE: kernel::procs::FaultResponse = kernel::procs::FaultResponse::Panic;
+
+// Manually setting the boot header section that contains the FCB header
+#[used]
+#[link_section = ".boot_hdr"]
+static BOOT_HDR: [u8; 8192] = boot_header::BOOT_HDR;
+
+// Force the emission of the `.apps` segment in the kernel elf image
+// NOTE: This will cause the kernel to overwrite any existing apps when flashed!
+#[used]
+#[link_section = ".app.hack"]
+static APP_HACK: u8 = 0;
```

Comment:
In https://github.com/tock/tock/pull/2001 we moved to a design that no longer relies on this Hack, could you incorporate that approach here as well?

Reply:
Done.